haskell/private/path_utils.bzl

```diff
@@ -318,3 +318,103 @@ def copy_all(ctx, srcs, dest):
             command = "mkdir -p {dest} && cp -L -R \"$@\" {dest}".format(dest = dest.path),
             arguments = [args],
         )
+
+def parse_pattern(pattern_str):
+    """Parses a string label pattern.
```

Comment:
If we move this to the general utilities we should link to https://github.com/bazelbuild/bazel/issues/7763 and require that every piece of code using this has to be an experimental API.